sdk/src/azure/core/az_json_writer.c

```diff
@@ -135,6 +135,11 @@ static AZ_NODISCARD bool _az_is_appending_value_valid(az_json_writer const* json
 }
 
 #ifndef AZ_NO_PRECONDITION_CHECKING
+// Declared to avoid warning C28278: Function appears with no prototype in scope.  Only limited
```

Comment:
Interesting. Looks like these are needed because they are only defined/used within `AZ_NO_PRECONDITION_CHECKING` #if/def blocks.

Reply:
No, I tried to declare them first, and it didn't help. But I think, code analysis tool still has bugs, so it is unfixable with reasonable costs. Maybe, we can make it go away if we declare the function as non-static, but I don't think it's worth it. Eventually, code analysis will probably make its analysis tool smarter so even lack of forward declaration won't be a problem for it.

Reply:
> Maybe, we can make it go away if we declare the function as non-static, but I don't think it's worth it. 

Agreed.

>No, I tried to declare them first, and it didn't help. But I think, code analysis tool still has bugs, so it is unfixable with reasonable costs.

FYI, @barcharcraz